libcontainer/internal/unix-utils/unix.go

```diff
@@ -0,0 +1,29 @@
+package unixutils
+
+import (
+	"errors"
+
+	"golang.org/x/sys/unix"
+)
+
+// RetryOnEINTR takes a function that returns an error and calls it until it the error returned is
+// not EINTR.
+func RetryOnEINTR(fn func() error) error {
+	var err error
```

Comment:
nit: no need to instantiate this, you can use `err := ` inside the loop, or have a named return variable.